sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -272,6 +312,10 @@ static az_result connecting(az_event_policy* me, az_event event)
       break;
     }
 
+    case AZ_MQTT5_EVENT_DISCONNECT_RSP:
+      _az_RETURN_IF_FAILED(_az_hfsm_transition_peer((_az_hfsm*)me, connecting, faulted));
```

Comment:
Can you add a comment here as well why this goes to faulted?

Reply:
For the comment: this is caused by protocol violation. Client sent `CONNECT` (Idle->Started action) and server replied with `DISCONNECT`.